fvm/src/kernel/default.rs

```diff
@@ -511,14 +511,26 @@ where
                 .on_verify_aggregate_signature(num_signers, plaintexts_concat.len()),
         )?;
 
-        let plaintexts: Vec<&[u8]> = plaintext_lens
+        let mut offset: usize = 0;
+        let plaintexts = plaintext_lens
             .iter()
-            .map(|len| {
-                let (plaintext, rem) = plaintexts_concat.split_at(*len as usize);
-                plaintexts_concat = rem;
-                plaintext
+            .map(|&len| {
+                let start = offset;
+                offset = start
+                    .checked_add(len as usize)
+                    .context("invalid bls signature length")
```

Comment:
Should this be `invalid bls plaintext length`?

Reply:
Yep!